public/app/features/dashboard/state/getPanelPluginToMigrateTo.ts

```diff
@@ -7,12 +7,30 @@ export function getPanelPluginToMigrateTo(panel: any, forceMigration?: boolean):
     return autoMigrateRemovedPanelPlugins[panel.type];
   }
 
+  const isUrlFeatureFlagEnabled = (featureName: string) => {
+    const flag = '__feature.' + featureName;
+
+    const urlParams = new URLSearchParams(window.location.search);
+    const featureFlagValue = urlParams.get(flag);
+
+    return featureFlagValue === 'true' || featureFlagValue === '';
+  };
+
   // Auto-migrate old angular panels
   const shouldMigrateAllAngularPanels =
-    forceMigration || !config.angularSupportEnabled || config.featureToggles.autoMigrateOldPanels;
+    forceMigration ||
+    !config.angularSupportEnabled ||
+    config.featureToggles.autoMigrateOldPanels ||
+    isUrlFeatureFlagEnabled('autoMigrateOldPanels') ||
+    isUrlFeatureFlagEnabled('disableAngular');
 
   // Graph needs special logic as it can be migrated to multiple panels
-  if (panel.type === 'graph' && (shouldMigrateAllAngularPanels || config.featureToggles.autoMigrateGraphPanel)) {
+  if (
+    panel.type === 'graph' &&
+    (shouldMigrateAllAngularPanels ||
+      config.featureToggles.autoMigrateGraphPanel ||
+      isUrlFeatureFlagEnabled('autoMigrateGraphPanel'))
```

Comment:
Does this ever happen via the "try migration" button in the banner? Or does that always set the `autoMigrateOldPanels` feature toggle to true? In which case this code is meant to support the manual case of testing granular panel by panel migration via url query params?

Reply:
Yes this gets triggered via "try migration", it's not setting/allowing any feature flags and it's just checking the url for the ff and migrates only the current dashboard.